presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -491,6 +491,15 @@ public PlanOptimizers(
         // Optimizers above this don't understand local exchanges, so be careful moving this.
         builder.add(new AddLocalExchanges(metadata, sqlParser));
 
+        // TODO: move this before optimization if possible!!
+        // Replace all expressions with row expressions
+        builder.add(new IterativeOptimizer(
+                ruleStats,
+                statsCalculator,
+                costCalculator,
+                new TranslateExpressions(metadata, sqlParser).rules()));
+        // After this point, all planNodes should not contain OriginalExpression
```

Comment:
`ApplyConnectorOptimization` needs to be moved up as well.

Reply:
I think it is arguable `ApplyConnectorOptimization` should always happen as early as possible. 
In the case of pushdown, it should happen after most optimizers.
In the case of federation, it should happen at logical planning phase.

In anyway, we should consider the interaction between the connector provided rules and optimization rules. So it is not as simple as moving up `TransalteExpressions`.